projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -92,7 +92,7 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
      * ```
      */
     @Input()
-    public igxForOf: any[] | null;
+    public igxForOf: U&T[] | null;
```

Comment:
This is how Angular does it. I guess there's a reason:

https://github.com/angular/angular/blob/4d18ba1af78dae2ba14f45ed092a4c55335edd9c/packages/common/src/directives/ng_for_of.ts#L140-L149

Reply:
The intersection helps in the typing for the single item in the `for...of`, otherwise it does not appear typed.
Here's the data:
![image](https://github.com/IgniteUI/igniteui-angular/assets/3768136/5488b594-056d-4d40-8464-88de33da6d8e)

If we set only `U` type on the `igxForOf` input:
![image](https://github.com/IgniteUI/igniteui-angular/assets/3768136/8c0ae460-8feb-4892-8690-0fcc20cfa4f6)

If we set it as `T[]` only, `U` ends up unused and appears to be used in the Angular ngFor, for the context that provides fallback to the ngFor data value where you can get it in the template as:
![image](https://github.com/IgniteUI/igniteui-angular/assets/3768136/80b2d3f7-cf96-43a4-97b4-1f859629587e)

The last bit we currently do not support though, which we can log it as a feature request.